src/chronify/store.py

```diff
@@ -232,29 +355,125 @@ def load_table(self, path: Path, schema: TableSchema) -> None:
             msg = "Only Parquet files are currently supported: {path=}"
             raise NotImplementedError(msg)
 
-        self.create_view_from_parquet(schema.name, path)
+        self.create_view_from_parquet(schema, path)
         try:
             self._check_table_schema(schema)
             self._check_timestamps(schema)
         except InvalidTable:
-            with self._engine.begin() as conn:
-                conn.execute(text(f"DROP VIEW {schema.name}"))
+            self.drop_view(schema.name)
             raise
 
+    def delete_rows(self, name: str, time_array_id_columns: dict[str, Any]) -> None:
+        """Delete all rows matching the time_array_id_columns."""
+        table = self.get_table(name)
+        if table is None:
+            msg = f"No table with {name=} is stored"
+            raise TableNotStored(msg)
+
+        if not time_array_id_columns:
+            msg = "time_array_id_columns cannot be empty"
+            raise InvalidParameter(msg)
+
+        stmt = None
```

Comment:
why can't `stmt = delete(table)` here? time_array_id_columns cannot be empty.

Reply:
Sure, good point